### PR TITLE
CPLP-1904 add company identifier to registration data

### PR DESCRIPTION
**Pull request by ntruchsess**
implements https://jira.catena-x.net/browse/CPLP-1904

### CODE REVIEW

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs [hunk 1]
```diff
@@ -227,4 +227,39 @@ public void AttachAndModifyCompanyApplication(Guid companyApplicationId, Action<
                     companyApplication.Invitations.Select(x => new InvitedCompanyUserData(x.CompanyUserId, x.CompanyUser!.Firstname, x.CompanyUser.Lastname, x.CompanyUser.Email))))
             .AsNoTracking()
             .SingleOrDefaultAsync();
+
+    public Task<(bool IsValidApplicationId, bool IsSameCompanyUser, RegistrationData? Data)> GetRegistrationDataUntrackedAsync(Guid applicationId, string iamUserId, IEnumerable<DocumentTypeId> documentTypes) =>
+        _dbContext.CompanyApplications
+            .AsNoTracking()
+            .AsSplitQuery()
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs:234]: yes - as the resultset of the database is always a table as soon the data logically spawns more than two dimensions the size will grow big as the numbers of records being returned by each dimension do multiply.
(any subselect that has multiple results will add another dimension)

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs [hunk 2]
```diff
@@ -227,4 +227,39 @@ public void AttachAndModifyCompanyApplication(Guid companyApplicationId, Action<
...
+                    x.Company.Shortname,
+                    x.Company.Address!.City,
+                    x.Company.Address.Region,
+                    x.Company.Address.Streetadditional,
+                    x.Company.Address.Streetname,
+                    x.Company.Address.Streetnumber,
+                    x.Company.Address.Zipcode,
+                    x.Company.Address.CountryAlpha2Code,
+                    x.Company.Address.Country!.CountryNameDe,
+                    x.Company.CompanyAssignedRoles!.Select(companyAssignedRole => companyAssignedRole.CompanyRoleId),
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs:257]: fixed

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs [hunk 3]
```diff
@@ -227,4 +227,39 @@ public void AttachAndModifyCompanyApplication(Guid companyApplicationId, Action<
...
+                    x.Company.Consents.Where(consent => consent.ConsentStatusId == PortalBackend.PortalEntities.Enums.ConsentStatusId.ACTIVE)
+                        .Select(consent => new ValueTuple<Guid,ConsentStatusId>(
+                            consent.AgreementId, consent.ConsentStatusId)),
+                    x.Company.CompanyUsers.SelectMany(companyUser => companyUser!.Documents!.Where(document=>documentTypes.Contains(document.DocumentTypeId)).Select(document => document.DocumentName)),
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs:261]: fixed

File: src/registration/Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs [hunk 4]
```diff
@@ -602,14 +602,39 @@ public async Task<int> SetInvitationStatusAsync(string iamUserId)
...
+            data.Region,
+            data.StreetAdditional,
+            data.StreetName,
+            data.StreetNumber,
+            data.ZipCode,
+            data.CountryAlpha2Code,
+            data.CountryDe,
+            data.CompanyRoleIds,
+            data.AgreementConsentStatuses.Select(consentStatus => new AgreementConsentStatusForRegistrationData(consentStatus.AgreementId, consentStatus.ConsentStatusId)),
+            data.DocumentNames.Select(name => new RegistrationDocumentNames(name)),
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/registration/Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs:635]: agree - this is from preexisting code. I didn't want to change the api-endpoint in an incompatible way. I'll clarify with @Julia whether this is really intended

File: tests/portalbackend/PortalBackend.DBAccess.Tests/ApplicationRepositoryTest.cs [hunk 5]
```diff
@@ -64,6 +65,28 @@ public async Task GetCompanyUserRoleWithAdressUntrackedAsync_WithExistingEntry_R
         results.InvitedCompanyUserData.Should().ContainSingle(u => u.FirstName == "Test User 2" && u.LastName == "cx-admin-2" && u.Email == "tester.user2@test.de");
     }
 
+    #endregion GetRegistrationDataUntrackedAsync
+
+    #region 
+    [Fact]
+    public async Task GetRegistrationDataUntrackedAsync_WithApplicationIdAndDocumentType_ReturnsExpectedResult()
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply tests/portalbackend/PortalBackend.DBAccess.Tests/ApplicationRepositoryTest.cs:72]: implemented

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs:234]: thanks for the explination

**Review by Phil91 — APPROVED** [review]

**Pull request merged**